internal/background/availability_queue.go

```diff
@@ -60,6 +60,8 @@ func sendAvailabilityRequestMessages(ctx context.Context, batchSize int, tickDur
 				send(ctx, messageBuffer...)
 				messageBuffer = messageBuffer[:0]
 			}
+
+			break
```

Comment:
sorry it took long, but I struggled to find time to verify a hunch that the `break` might be breaking the select and it appears it does: https://stackoverflow.com/questions/25469682/break-out-of-select-loop

So I guess we want to `return` here?
```suggestion
			return
```

Reply:
Don't we want a `default` option as well? 
Correct me if I am wrong but if we get a signal to terminate won't we want to break the select block as well?

Reply:
return should break both `select` and `for` :)
default would cause the select not wait for the signals and would go to the default right away I belive :)

Reply:
Oh nice find, I haven't actually tested this. Shame on me.